consul/src/main/java/com/linecorp/armeria/internal/consul/HealthClient.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.HttpStatus;
+
+/**
+ * Health client supports below APIs.
+ * {@code HealthClient} is responsible for endpoint of Consul API:
+ * {@code `/health`}(https://www.consul.io/api/health.html)
+ *
+ * <p>GET /health/node/:node
+ * GET /health/checks/:service
+ * GET /health/service/:service?passing
+ * GET /health/connect/:service
+ * GET /health/state/:state
+ */
+public class HealthClient {
```

Comment:
Remove public? 
```suggestion
final class HealthClient {
```

Reply:
Ok, I changed all inside internal package except `Check` and `ConsulClient`.